app/jobs/notifications/remove_each_job.rb

```diff
@@ -0,0 +1,15 @@
+module Notifications
+  class RemoveEachJob < ApplicationJob
+    queue_as :remove_each_notifications
+
+    def perform(notifiable_collection_id, action, service = Notifications::RemoveEach)
+      notifiable_collection = notifiable_collection_id.each_with_object([]) do |id, obj|
+        obj << Mention.find_by(id: id)
```

Comment:
I don't think this is very efficient, if the caller sends 100 ids, this is going to issue 100 queries.

You can just call `notifiable_collection = Mention.where(id: notificable_collection_ids])` and then ask the DB if there are any: `return unless notifiable_collection.any?` or `return unless notifiable_collection.present?` which eagerly loads the objects you're going to send to the collection in the service call.

My suggestion is to use `any?` here and then use something like `notifiable_collection.select(:id)` in the service call so you only load the `id` field from the DB.

Still, one or two queries instead of N...

Reply:
Oops, Thank you for noticing. I have removed that horrible block. I need to pass objects to the service, so I did not use the select(:id)

Reply:
`select(:id)` loads objects, it's different from `pluck(:id)`. Instead of `SELECT *` issues a `SELECT id` but it still builds ActiveRecord objects.

You can see how it works here: https://api.rubyonrails.org/classes/ActiveRecord/Associations/CollectionProxy.html#method-i-select

Reply:
Thank you for the tip. Due to the methods not only for Mention anymore, I had to refactor in another way.